### PR TITLE
[KYUUBI #4142][DOCS] Update the docs link in README

### PR DESCRIPTION
**Pull request by runzhliu**
### _Why are the changes needed?_

Some links of the website are not available.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ and others would not be possible without your help.
 ## Online Documentation
 
 Since Kyuubi 1.3.0-incubating, the Kyuubi online documentation is hosted by [https://kyuubi.apache.org/](https://kyuubi.apache.org/).
-You can find the latest Kyuubi documentation on [this web page](https://kyuubi.apache.org/docs/latest/).
+You can find the latest Kyuubi documentation on [this web page](https://kyuubi.apache.org/).
```
Comment (pan3793) [inline README.md:105]: How about pointing this to `https://kyuubi.readthedocs.io/en/master/`? cc @yaooqinn

Comment (yaooqinn) [reply README.md:105]: LGTM

Comment (pan3793) [reply README.md:105]: @runzhliu would you please update it?

Comment (runzhliu) [reply README.md:105]: this [link](https://kyuubi.readthedocs.io/en/master/) also seems to be unavailable, and always be redirected to https://kyuubi.apache.org/en/master/ @pan3793

Comment (pan3793) [reply README.md:105]: we are investigating the issue

Comment (yaooqinn) [reply README.md:105]: fixed

File: README.md [hunk 2]
```diff
@@ -102,7 +102,7 @@ and others would not be possible without your help.
 ## Online Documentation
 
 Since Kyuubi 1.3.0-incubating, the Kyuubi online documentation is hosted by [https://kyuubi.apache.org/](https://kyuubi.apache.org/).
-You can find the latest Kyuubi documentation on [this web page](https://kyuubi.apache.org/docs/latest/).
+You can find the latest Kyuubi documentation on [this web page](https://kyuubi.readthedocs.io/en/master/).
 For 1.2 and earlier versions, please check the [Readthedocs](https://kyuubi.readthedocs.io/en/v1.2.0/) directly.
 
 ## Quick Start
```
Comment (cfmcgrady) [inline README.md:108]: the link of `Getting Started` is unavailable, could you update it as well?

Comment (runzhliu) [reply README.md:108]: done. @cfmcgrady

**Review by cfmcgrady — APPROVED** [review]

**Review by bowenliang123 — APPROVED** [review]

**Pull request closed without merging**